infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/server/external/TraefikGatewayRouteConfigGenerator.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.kubernetes.server.external;
+
+import static com.fasterxml.jackson.dataformat.yaml.YAMLGenerator.Feature.WRITE_DOC_START_MARKER;
+
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.fasterxml.jackson.dataformat.yaml.YAMLGenerator;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * Traefik configuration for single route looks like this (values in {} are parameters of {@link
+ * GatewayRouteConfigGenerator#generate(String, String, String)} method):
+ *
+ * <pre>
+ * http:
+ *   routers:
+ *     {name}:
+ *       rule: "PathPrefix(`{path}`)"
+ *       service: {name}
+ *       middlewares:
+ *       - "{name}"
+ *       priority: 100
+ *   services:
+ *     {name}:
+ *       loadBalancer:
+ *         servers:
+ *         - url: '{serviceUrl}'
+ *   middlewares:
+ *     {name}:
+ *       stripPrefix:
+ *         prefixes:
+ *         - "{path}"
+ * </pre>
+ */
+public class TraefikGatewayRouteConfigGenerator implements GatewayRouteConfigGenerator {
+
+  /**
+   * Generates Traefik specific configuration for single service.
+   *
+   * @param name name of the service
+   * @param serviceUrl url of service we want to route to
+   * @param path path to route and strip
+   * @return traefik service route config
+   */
+  @Override
+  public Map<String, String> generate(String name, String serviceUrl, String path) {
+    StringWriter sw = new StringWriter();
+
+    try {
+      YAMLGenerator generator =
```

Comment:
Not sure about using `YAMLGenerator`. We could use `ObjectMapper`, but then we would need to create all objects in configuration yaml or use multiple levels of `Map<String, Map<String, Map<...`.